File: frontend/src/types/patient.ts
export interface Patient {
  id: number;
  age: number;
  sex: string;
  systolic_bp: number;
  diastolic_bp: number;
  heart_rate: number;
  max_hr: number;
  cholesterol: number;
  st_depression: number;
  exercise_angina: boolean;
  hrv: number;
  ejection_fraction: number;
  stroke_volume: number;
  cardiac_output: number;
  disease_severity: number;
}

export interface PatientBrief {
  id: number;
  label: string;
  age: number;
  sex: string;
}

export interface AIAnalysis {
  observation: string;
  risk_factor: string;
  recommendation: string;
  risk_score: number;
  risk_level: 'low' | 'moderate' | 'high' | 'unknown';
}

File: backend/app/models/patient.py
from pydantic import BaseModel
from typing import Optional, List

class PatientBase(BaseModel):
    age: int
    sex: str
    systolic_bp: int
    diastolic_bp: int
    heart_rate: int
    cholesterol: int

class PatientResponse(PatientBase):
    id: int
    max_hr: int
    st_depression: float
    exercise_angina: bool
    hrv: int
    ejection_fraction: int
    stroke_volume: int
    cardiac_output: float
    disease_severity: int

class PatientBrief(BaseModel):
    id: int
    label: str
    age: int
    sex: str

class VitalsUpdate(BaseModel):
    systolic_bp: Optional[int] = None
    diastolic_bp: Optional[int] = None
    heart_rate: Optional[int] = None
    cholesterol: Optional[int] = None

class AIAnalysisRequest(BaseModel):
    systolic_bp: int
    diastolic_bp: int
    heart_rate: int
    ejection_fraction: int
    st_depression: float
    cholesterol: int

class AIAnalysisResponse(BaseModel):
    observation: str
    risk_factor: str
    recommendation: str
    risk_score: int
    risk_level: str  # "low", "moderate", "high"
